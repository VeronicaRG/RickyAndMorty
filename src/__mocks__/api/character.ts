export const MOCK_CHARACTER = {
  gender: 'Male',
  id: '1',
  image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
  name: 'Rick Sanchez',
  species: 'Human',
  status: 'Alive',
};

export const MOCK_CHARACTER_DETAILS = {
  data: {
    character: {
      id: '1',
      name: 'Rick Sanchez',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      gender: 'Male',
      status: 'Alive',
      species: 'Human',
      origin: {
        name: 'Earth (C-137)',
        __typename: 'Location',
      },
      location: {
        name: 'Citadel of Ricks',
        __typename: 'Location',
      },
      episode: [
        {
          id: '1',
          air_date: 'December 2, 2013',
          episode: 'S01E01',
          created: '2017-11-10T12:56:33.798Z',
          __typename: 'Episode',
        },
        {
          id: '2',
          air_date: 'December 9, 2013',
          episode: 'S01E02',
          created: '2017-11-10T12:56:33.916Z',
          __typename: 'Episode',
        },
        {
          id: '3',
          air_date: 'December 16, 2013',
          episode: 'S01E03',
          created: '2017-11-10T12:56:34.022Z',
          __typename: 'Episode',
        },
        {
          id: '4',
          air_date: 'January 13, 2014',
          episode: 'S01E04',
          created: '2017-11-10T12:56:34.129Z',
          __typename: 'Episode',
        },
        {
          id: '5',
          air_date: 'January 20, 2014',
          episode: 'S01E05',
          created: '2017-11-10T12:56:34.236Z',
          __typename: 'Episode',
        },
        {
          id: '6',
          air_date: 'January 27, 2014',
          episode: 'S01E06',
          created: '2017-11-10T12:56:34.339Z',
          __typename: 'Episode',
        },
        {
          id: '7',
          air_date: 'March 10, 2014',
          episode: 'S01E07',
          created: '2017-11-10T12:56:34.441Z',
          __typename: 'Episode',
        },
        {
          id: '8',
          air_date: 'March 17, 2014',
          episode: 'S01E08',
          created: '2017-11-10T12:56:34.543Z',
          __typename: 'Episode',
        },
        {
          id: '9',
          air_date: 'March 24, 2014',
          episode: 'S01E09',
          created: '2017-11-10T12:56:34.645Z',
          __typename: 'Episode',
        },
        {
          id: '10',
          air_date: 'April 7, 2014',
          episode: 'S01E10',
          created: '2017-11-10T12:56:34.747Z',
          __typename: 'Episode',
        },
        {
          id: '11',
          air_date: 'April 14, 2014',
          episode: 'S01E11',
          created: '2017-11-10T12:56:34.850Z',
          __typename: 'Episode',
        },
        {
          id: '12',
          air_date: 'July 26, 2015',
          episode: 'S02E01',
          created: '2017-11-10T12:56:34.953Z',
          __typename: 'Episode',
        },
        {
          id: '13',
          air_date: 'August 2, 2015',
          episode: 'S02E02',
          created: '2017-11-10T12:56:35.055Z',
          __typename: 'Episode',
        },
        {
          id: '14',
          air_date: 'August 9, 2015',
          episode: 'S02E03',
          created: '2017-11-10T12:56:35.158Z',
          __typename: 'Episode',
        },
        {
          id: '15',
          air_date: 'August 16, 2015',
          episode: 'S02E04',
          created: '2017-11-10T12:56:35.261Z',
          __typename: 'Episode',
        },
        {
          id: '16',
          air_date: 'August 23, 2015',
          episode: 'S02E05',
          created: '2017-11-10T12:56:35.364Z',
          __typename: 'Episode',
        },
        {
          id: '17',
          air_date: 'August 30, 2015',
          episode: 'S02E06',
          created: '2017-11-10T12:56:35.467Z',
          __typename: 'Episode',
        },
        {
          id: '18',
          air_date: 'September 13, 2015',
          episode: 'S02E07',
          created: '2017-11-10T12:56:35.569Z',
          __typename: 'Episode',
        },
        {
          id: '19',
          air_date: 'September 20, 2015',
          episode: 'S02E08',
          created: '2017-11-10T12:56:35.669Z',
          __typename: 'Episode',
        },
        {
          id: '20',
          air_date: 'September 27, 2015',
          episode: 'S02E09',
          created: '2017-11-10T12:56:35.772Z',
          __typename: 'Episode',
        },
        {
          id: '21',
          air_date: 'October 4, 2015',
          episode: 'S02E10',
          created: '2017-11-10T12:56:35.875Z',
          __typename: 'Episode',
        },
        {
          id: '22',
          air_date: 'April 1, 2017',
          episode: 'S03E01',
          created: '2017-11-10T12:56:35.983Z',
          __typename: 'Episode',
        },
        {
          id: '23',
          air_date: 'July 30, 2017',
          episode: 'S03E02',
          created: '2017-11-10T12:56:36.100Z',
          __typename: 'Episode',
        },
        {
          id: '24',
          air_date: 'August 6, 2017',
          episode: 'S03E03',
          created: '2017-11-10T12:56:36.206Z',
          __typename: 'Episode',
        },
        {
          id: '25',
          air_date: 'August 13, 2017',
          episode: 'S03E04',
          created: '2017-11-10T12:56:36.310Z',
          __typename: 'Episode',
        },
        {
          id: '26',
          air_date: 'August 20, 2017',
          episode: 'S03E05',
          created: '2017-11-10T12:56:36.413Z',
          __typename: 'Episode',
        },
        {
          id: '27',
          air_date: 'August 27, 2017',
          episode: 'S03E06',
          created: '2017-11-10T12:56:36.515Z',
          __typename: 'Episode',
        },
        {
          id: '28',
          air_date: 'September 10, 2017',
          episode: 'S03E07',
          created: '2017-11-10T12:56:36.618Z',
          __typename: 'Episode',
        },
        {
          id: '29',
          air_date: 'September 17, 2017',
          episode: 'S03E08',
          created: '2017-11-10T12:56:36.726Z',
          __typename: 'Episode',
        },
        {
          id: '30',
          air_date: 'September 24, 2017',
          episode: 'S03E09',
          created: '2017-11-10T12:56:36.828Z',
          __typename: 'Episode',
        },
        {
          id: '31',
          air_date: 'October 1, 2017',
          episode: 'S03E10',
          created: '2017-11-10T12:56:36.929Z',
          __typename: 'Episode',
        },
        {
          id: '32',
          air_date: 'November 10, 2019',
          episode: 'S04E01',
          created: '2020-04-30T06:52:04.495Z',
          __typename: 'Episode',
        },
        {
          id: '33',
          air_date: 'November 17, 2019',
          episode: 'S04E02',
          created: '2020-04-30T06:52:04.498Z',
          __typename: 'Episode',
        },
        {
          id: '34',
          air_date: 'November 24, 2019',
          episode: 'S04E03',
          created: '2020-04-30T06:52:04.498Z',
          __typename: 'Episode',
        },
        {
          id: '35',
          air_date: 'December 8, 2019',
          episode: 'S04E04',
          created: '2020-04-30T06:52:04.498Z',
          __typename: 'Episode',
        },
        {
          id: '36',
          air_date: 'December 15, 2019',
          episode: 'S04E05',
          created: '2020-04-30T06:52:04.499Z',
          __typename: 'Episode',
        },
        {
          id: '37',
          air_date: 'May 3, 2020',
          episode: 'S04E06',
          created: '2020-08-06T05:44:21.422Z',
          __typename: 'Episode',
        },
        {
          id: '38',
          air_date: 'May 10, 2020',
          episode: 'S04E07',
          created: '2020-08-06T05:49:40.563Z',
          __typename: 'Episode',
        },
        {
          id: '39',
          air_date: 'May 17, 2020',
          episode: 'S04E08',
          created: '2020-08-06T05:51:07.419Z',
          __typename: 'Episode',
        },
        {
          id: '40',
          air_date: 'May 24, 2020',
          episode: 'S04E09',
          created: '2020-08-06T05:51:25.458Z',
          __typename: 'Episode',
        },
        {
          id: '41',
          air_date: 'May 31, 2020',
          episode: 'S04E10',
          created: '2020-08-06T05:51:52.079Z',
          __typename: 'Episode',
        },
        {
          id: '42',
          air_date: 'June 20, 2021',
          episode: 'S05E01',
          created: '2021-10-15T17:00:24.099Z',
          __typename: 'Episode',
        },
        {
          id: '43',
          air_date: 'June 27, 2021',
          episode: 'S05E02',
          created: '2021-10-15T17:00:24.102Z',
          __typename: 'Episode',
        },
        {
          id: '44',
          air_date: 'July 4, 2021',
          episode: 'S05E03',
          created: '2021-10-15T17:00:24.102Z',
          __typename: 'Episode',
        },
        {
          id: '45',
          air_date: 'July 11, 2021',
          episode: 'S05E04',
          created: '2021-10-15T17:00:24.103Z',
          __typename: 'Episode',
        },
        {
          id: '46',
          air_date: 'July 18, 2021',
          episode: 'S05E05',
          created: '2021-10-15T17:00:24.103Z',
          __typename: 'Episode',
        },
        {
          id: '47',
          air_date: 'July 25, 2021',
          episode: 'S05E06',
          created: '2021-10-15T17:00:24.104Z',
          __typename: 'Episode',
        },
        {
          id: '48',
          air_date: 'August 1, 2021',
          episode: 'S05E07',
          created: '2021-10-15T17:00:24.104Z',
          __typename: 'Episode',
        },
        {
          id: '49',
          air_date: 'August 8, 2021',
          episode: 'S05E08',
          created: '2021-10-15T17:00:24.104Z',
          __typename: 'Episode',
        },
        {
          id: '50',
          air_date: 'September 5, 2021',
          episode: 'S05E09',
          created: '2021-10-15T17:00:24.105Z',
          __typename: 'Episode',
        },
        {
          id: '51',
          air_date: 'September 5, 2021',
          episode: 'S05E10',
          created: '2021-10-15T17:00:24.105Z',
          __typename: 'Episode',
        },
      ],
      __typename: 'Character',
    },
  },
};

export const MOCK_GET_CHARACTERS_PAGE_1 = {
  data: {
    characters: {
      results: [
        {
          id: '1',
          name: 'Rick Sanchez',
          image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '2',
          name: 'Morty Smith',
          image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '3',
          name: 'Summer Smith',
          image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg',
          species: 'Human',
          gender: 'Female',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '4',
          name: 'Beth Smith',
          image: 'https://rickandmortyapi.com/api/character/avatar/4.jpeg',
          species: 'Human',
          gender: 'Female',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '5',
          name: 'Jerry Smith',
          image: 'https://rickandmortyapi.com/api/character/avatar/5.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '6',
          name: 'Abadango Cluster Princess',
          image: 'https://rickandmortyapi.com/api/character/avatar/6.jpeg',
          species: 'Alien',
          gender: 'Female',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '7',
          name: 'Abradolf Lincler',
          image: 'https://rickandmortyapi.com/api/character/avatar/7.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'unknown',
          __typename: 'Character',
        },
        {
          id: '8',
          name: 'Adjudicator Rick',
          image: 'https://rickandmortyapi.com/api/character/avatar/8.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '9',
          name: 'Agency Director',
          image: 'https://rickandmortyapi.com/api/character/avatar/9.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '10',
          name: 'Alan Rails',
          image: 'https://rickandmortyapi.com/api/character/avatar/10.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '11',
          name: 'Albert Einstein',
          image: 'https://rickandmortyapi.com/api/character/avatar/11.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '12',
          name: 'Alexander',
          image: 'https://rickandmortyapi.com/api/character/avatar/12.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '13',
          name: 'Alien Googah',
          image: 'https://rickandmortyapi.com/api/character/avatar/13.jpeg',
          species: 'Alien',
          gender: 'unknown',
          status: 'unknown',
          __typename: 'Character',
        },
        {
          id: '14',
          name: 'Alien Morty',
          image: 'https://rickandmortyapi.com/api/character/avatar/14.jpeg',
          species: 'Alien',
          gender: 'Male',
          status: 'unknown',
          __typename: 'Character',
        },
        {
          id: '15',
          name: 'Alien Rick',
          image: 'https://rickandmortyapi.com/api/character/avatar/15.jpeg',
          species: 'Alien',
          gender: 'Male',
          status: 'unknown',
          __typename: 'Character',
        },
        {
          id: '16',
          name: 'Amish Cyborg',
          image: 'https://rickandmortyapi.com/api/character/avatar/16.jpeg',
          species: 'Alien',
          gender: 'Male',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '17',
          name: 'Annie',
          image: 'https://rickandmortyapi.com/api/character/avatar/17.jpeg',
          species: 'Human',
          gender: 'Female',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '18',
          name: 'Antenna Morty',
          image: 'https://rickandmortyapi.com/api/character/avatar/18.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '19',
          name: 'Antenna Rick',
          image: 'https://rickandmortyapi.com/api/character/avatar/19.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'unknown',
          __typename: 'Character',
        },
        {
          id: '20',
          name: 'Ants in my Eyes Johnson',
          image: 'https://rickandmortyapi.com/api/character/avatar/20.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'unknown',
          __typename: 'Character',
        },
      ],
      info: {
        next: 2,
        __typename: 'Info',
      },
      __typename: 'Characters',
    },
  },
};

export const MOCK_GET_CHARACTERS_PAGE_2 = {
  data: {
    characters: {
      results: [
        {
          id: '21',
          name: 'Aqua Morty',
          image: 'https://rickandmortyapi.com/api/character/avatar/21.jpeg',
          species: 'Humanoid',
          gender: 'Male',
          status: 'unknown',
          __typename: 'Character',
        },
        {
          id: '22',
          name: 'Aqua Rick',
          image: 'https://rickandmortyapi.com/api/character/avatar/22.jpeg',
          species: 'Humanoid',
          gender: 'Male',
          status: 'unknown',
          __typename: 'Character',
        },
        {
          id: '23',
          name: 'Arcade Alien',
          image: 'https://rickandmortyapi.com/api/character/avatar/23.jpeg',
          species: 'Alien',
          gender: 'Male',
          status: 'unknown',
          __typename: 'Character',
        },
        {
          id: '24',
          name: 'Armagheadon',
          image: 'https://rickandmortyapi.com/api/character/avatar/24.jpeg',
          species: 'Alien',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '25',
          name: 'Armothy',
          image: 'https://rickandmortyapi.com/api/character/avatar/25.jpeg',
          species: 'unknown',
          gender: 'Male',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '26',
          name: 'Arthricia',
          image: 'https://rickandmortyapi.com/api/character/avatar/26.jpeg',
          species: 'Alien',
          gender: 'Female',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '27',
          name: 'Artist Morty',
          image: 'https://rickandmortyapi.com/api/character/avatar/27.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '28',
          name: 'Attila Starwar',
          image: 'https://rickandmortyapi.com/api/character/avatar/28.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '29',
          name: 'Baby Legs',
          image: 'https://rickandmortyapi.com/api/character/avatar/29.jpeg',
          species: 'Human',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '30',
          name: 'Baby Poopybutthole',
          image: 'https://rickandmortyapi.com/api/character/avatar/30.jpeg',
          species: 'Poopybutthole',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '31',
          name: 'Baby Wizard',
          image: 'https://rickandmortyapi.com/api/character/avatar/31.jpeg',
          species: 'Alien',
          gender: 'Male',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '32',
          name: 'Bearded Lady',
          image: 'https://rickandmortyapi.com/api/character/avatar/32.jpeg',
          species: 'Alien',
          gender: 'Female',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '33',
          name: 'Beebo',
          image: 'https://rickandmortyapi.com/api/character/avatar/33.jpeg',
          species: 'Alien',
          gender: 'Male',
          status: 'Dead',
          __typename: 'Character',
        },
        {
          id: '34',
          name: 'Benjamin',
          image: 'https://rickandmortyapi.com/api/character/avatar/34.jpeg',
          species: 'Poopybutthole',
          gender: 'Male',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '35',
          name: 'Bepisian',
          image: 'https://rickandmortyapi.com/api/character/avatar/35.jpeg',
          species: 'Alien',
          gender: 'unknown',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '36',
          name: 'Beta-Seven',
          image: 'https://rickandmortyapi.com/api/character/avatar/36.jpeg',
          species: 'Alien',
          gender: 'unknown',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '37',
          name: 'Beth Sanchez',
          image: 'https://rickandmortyapi.com/api/character/avatar/37.jpeg',
          species: 'Human',
          gender: 'Female',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '38',
          name: 'Beth Smith',
          image: 'https://rickandmortyapi.com/api/character/avatar/38.jpeg',
          species: 'Human',
          gender: 'Female',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '39',
          name: 'Beth Smith',
          image: 'https://rickandmortyapi.com/api/character/avatar/39.jpeg',
          species: 'Human',
          gender: 'Female',
          status: 'Alive',
          __typename: 'Character',
        },
        {
          id: '40',
          name: "Beth's Mytholog",
          image: 'https://rickandmortyapi.com/api/character/avatar/40.jpeg',
          species: 'Mythological Creature',
          gender: 'Female',
          status: 'Dead',
          __typename: 'Character',
        },
      ],
      info: {
        next: 3,
        __typename: 'Info',
      },
      __typename: 'Characters',
    },
  },
};
