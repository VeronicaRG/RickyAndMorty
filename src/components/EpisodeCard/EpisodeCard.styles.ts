import { Icon } from '@components/Icon';
import { Row } from '@components/Row';
import { Dimensions } from 'react-native';
import FastImage from 'react-native-fast-image';
import styled from 'styled-components/native';

export const Card = styled.View<{ horizontal?: boolean }>`
  height: 140px;
  width: ${() => Dimensions.get('window').width - 32}px;
  background-color: #926f44;
  padding: 8px;
  flex-direction: row;
  margin-bottom: 16px;
  border-radius: 10px;
  border: 1px solid black;
`;

export const Image = styled(FastImage).attrs({
  resizeMode: 'cover',
})`
  flex: 1;
  border-radius: 10px;
  border: 1px solid black;
`;

export const InfoContainer = styled.View`
  flex: 1;
  margin: 0 6px;
`;

export const Label = styled.Text.attrs({
  numberOfLines: 1,
})`
  flex: 1;
  color: white;
  margin-left: 4px;
  font-size: 12px;
  font-weight: bold;
`;

export const Property = styled.Text<{ highlight?: boolean }>`
  color: white;
  margin: 4px 0;
  font-size: 10px;
  ${({ highlight }) => highlight && 'fontWeight: bold'}
`;

export const WrapperCharacterImage = styled.View`
  flex: 1;
  flex-direction: row;
`;

export const CharacterImage = styled(FastImage)`
  width: 22px;
  height: 22px;
  margin-right: 4px;
  align-self: center;
  border-radius: 12px;
  border: 1px solid black;
`;

export const PlusIcon = styled(Icon).attrs({
  name: 'plus',
  width: 20,
  height: 20,
})`
  align-self: center;
`;

export const IconWrapper = styled(Row)`
  flex: 1;
  margin: 6px 0;
  align-items: center;
  justify-content: space-between;
`;
