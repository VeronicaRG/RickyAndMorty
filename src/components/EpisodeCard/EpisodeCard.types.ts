import { Character, Episode, Maybe } from '@core-types/graphql';
import { TouchableOpacityProps } from 'react-native';

export type EpisodeCardProps = {
  episode: Partial<Omit<Episode, 'characters'>> & {
    image?: string;
    characters?: Partial<Maybe<Character>>[];
  };
} & TouchableOpacityProps;
