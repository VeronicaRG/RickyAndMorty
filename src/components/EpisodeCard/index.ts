export * from './EpisodeCard';
export * from './EpisodeCard.types';
