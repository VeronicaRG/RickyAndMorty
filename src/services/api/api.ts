import { ApolloClient, InMemoryCache, createHttpLink, from } from '@apollo/client';
import { API_ADDRESS } from '@env';

const httpLink = createHttpLink({ uri: API_ADDRESS });

export const apolloClient = new ApolloClient({
  link: from([httpLink]),
  cache: new InMemoryCache(),
});
