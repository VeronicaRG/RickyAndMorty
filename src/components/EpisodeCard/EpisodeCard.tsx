import React from 'react';

import * as S from './EpisodeCard.styles';
import { EpisodeCardProps } from './EpisodeCard.types';
import { convertAirDateToDateString, formatDate } from '@src/utils/formatters';
import { i18n } from '@src/languages';
import { FavoriteIcon } from '@components/FavoriteIcon';

export const EpisodeCard: React.FC<EpisodeCardProps> = ({
  episode,
  isFavorite,
  handlePressFavorite,
}) => {
  const { id, name, image, air_date, episode: episodeName, characters = [] } = episode;

  return (
    <S.Card testID={`episode-${id}`}>
      <S.Image source={{ uri: image }} />
      <S.InfoContainer>
        <S.IconWrapper>
          <S.Label>{name}</S.Label>
          <FavoriteIcon
            widthIcon={20}
            heightIcon={20}
            isFavorite={isFavorite}
            testID={`episode-${id}-`}
            handlePressFavorite={handlePressFavorite}
          />
        </S.IconWrapper>
        <S.Property>{episodeName}</S.Property>
        <S.Property>
          {formatDate({
            date: convertAirDateToDateString(air_date),
            format: '[day] [month] [year]',
          })}
        </S.Property>
        <S.Property highlight>{i18n.t('EpisodeDetails:characters')}</S.Property>
        <S.WrapperCharacterImage>
          {characters.slice(0, 6).map(character => (
            <S.CharacterImage key={character?.id} source={{ uri: character?.image! }} />
          ))}
          {characters.length > 7 && <S.PlusIcon />}
        </S.WrapperCharacterImage>
      </S.InfoContainer>
    </S.Card>
  );
};
