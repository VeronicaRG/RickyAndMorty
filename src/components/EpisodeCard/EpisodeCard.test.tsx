import React from 'react';
import { render } from '@testing-library/react-native';
import { EpisodeCard } from './EpisodeCard';
import { MOCK_EPISODE } from '@src/__mocks__/api/episode';

describe('Component: EpisodeCard', () => {
  it('should render component', () => {
    const { toJSON } = render(
      <EpisodeCard episode={MOCK_EPISODE} handlePressFavorite={() => {}} isFavorite />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
