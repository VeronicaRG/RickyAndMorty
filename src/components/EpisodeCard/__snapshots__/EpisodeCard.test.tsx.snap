// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: EpisodeCard should render component 1`] = `
<View
  style={
    {
      "backgroundColor": "#926f44",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderColor": "black",
      "borderStyle": "solid",
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 140,
      "marginBottom": 16,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
      "width": 718,
    }
  }
  testID="episode-1"
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "borderBottomLeftRadius": 10,
          "borderBottomRightRadius": 10,
          "borderColor": "black",
          "borderStyle": "solid",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "borderWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 0,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 0,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "marginBottom": 6,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 6,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          {
            "color": "white",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 12,
            "fontWeight": "bold",
            "marginLeft": 4,
          }
        }
      >
        Pilot
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="episode-1-favorite-icon-button"
      >
        <SvgMock
          color="#fff"
          height={20}
          testID="episode-1-favorite-icon-colorful"
          width={20}
        />
      </View>
    </View>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 10,
          "marginBottom": 4,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 4,
        }
      }
    >
      S01E01
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontSize": 10,
          "marginBottom": 4,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 4,
        }
      }
    >
      2 de dezembro de 2013
    </Text>
    <Text
      highlight={true}
      style={
        {
          "color": "white",
          "fontSize": 10,
          "fontWeight": "bold",
          "marginBottom": 4,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 4,
        }
      }
    >
      Principais personages
    </Text>
    <View
      style={
        {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "alignSelf": "center",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderColor": "black",
              "borderStyle": "solid",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderWidth": 1,
              "height": 22,
              "marginRight": 4,
              "width": 22,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "alignSelf": "center",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderColor": "black",
              "borderStyle": "solid",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderWidth": 1,
              "height": 22,
              "marginRight": 4,
              "width": 22,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "alignSelf": "center",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderColor": "black",
              "borderStyle": "solid",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderWidth": 1,
              "height": 22,
              "marginRight": 4,
              "width": 22,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "alignSelf": "center",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderColor": "black",
              "borderStyle": "solid",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderWidth": 1,
              "height": 22,
              "marginRight": 4,
              "width": 22,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "alignSelf": "center",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderColor": "black",
              "borderStyle": "solid",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderWidth": 1,
              "height": 22,
              "marginRight": 4,
              "width": 22,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "alignSelf": "center",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderColor": "black",
              "borderStyle": "solid",
              "borderTopLeftRadius": 12,
              "borderTopRightRadius": 12,
              "borderWidth": 1,
              "height": 22,
              "marginRight": 4,
              "width": 22,
            },
          ]
        }
      >
        <FastImageView
          defaultSource={null}
          resizeMode="cover"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <SvgMock
        color="#fff"
        height={20}
        style={
          {
            "alignSelf": "center",
          }
        }
        width={20}
      />
    </View>
  </View>
</View>
`;
